### PR TITLE
Removes debug assertion that was related to issue 796

### PR DESCRIPTION
**Pull request by kevaundray**

### CODE REVIEW

**Comment by fitzgen** [comment]
I don't think #796 is what you mean?

And I think this might effectively be a duplicate of https://github.com/bytecodealliance/wasmtime/pull/5553?

In any case, we don't want to simply remove assertions, we want to fix the underlying reason the assertion is failing. Therefore, we won't want to merge this PR in its current state.

**Comment by bjorn3** [comment]
It is issue 796 on the old cranelift repo before cranelift was merged into the wasmtime repo.

**Comment by kevaundray** [comment]
Happy to close, though maybe it's better to update the issue number on the PR?

File: cranelift/codegen/src/data_value.rs [hunk 1]
```diff
@@ -249,7 +249,7 @@ impl DataValue {
     ///
     /// Returns true if all bits are equal.
     ///
-    /// This behaviour is different from PartialEq for NaN floats.
+    /// This behavior is different from PartialEq for NaN floats.
```
Comment (jameysharp) [inline cranelift/codegen/src/data_value.rs:252]: I'd prefer not to have an unrelated change in this PR, and particularly not one that just changes spelling from British English to American English. Let's leave the Commonwealth folks alone. :laughing:

**Review by jameysharp — APPROVED** [review]
This is not related to #5553. I believe this was trying to assert that we only construct valid basic blocks. It might be useful to have a debug-assertion like that here. But I'm happy with just deleting it.

Allegedly, based on commit messages, the old issue 796 was fixed in 69d2f40, back in 2019. So deleting this debug-assertion seems long overdue.

I'm not convinced this assertion was even correct: if the previous instruction was a terminator, then it allows inserting another instruction after it, contrary to the definition of a terminating instruction. Also, I think it only checks when inserting before another instruction, in which case neither the previous nor the new instruction should be a terminator, but that's not what it tests.

Those mistakes could have been due to later changes, though. In particular, these days, `Opcode::is_branch` implies `Opcode::is_terminator`, which means I believe this panic is unreachable now.

I would prefer to remove the unrelated change from this commit before merging it, though.

File: cranelift/codegen/src/data_value.rs (hunk 1 above)
Comment (kevaundray) [reply cranelift/codegen/src/data_value.rs]: Done!

**Review by jameysharp — APPROVED** [review]
Great, thank you!
